### ISSUE TITLE
PEnG seeding - BaMM refinement workflow

### ISSUE THREAD

**Issue by croth1** [soedinglab/BaMM_webserver#26]
Maxi's current project is to replace the current BaMM workflow with a two step process:

1) Seeding phase: runs PEnG-motif to find seeds for BaMM motif
2) Refinement phase: the user selects the peng seeds to be optimized by bamm

Maxi has uploaded his work recently (Thanks a lot!). Here's my non-comprehensive list of things we should do before we can run the new version on the offical BaMM webserver:

# Planned Changes
## Critical Changes
- [x] use JobMeta table for BaMM jobs to avoid JobID clashes
- [x] rebase upon Anja’s latest version
- [x] make PEnG stage runnable with example data
- [ ] update PEnG to latest version (Christian) 
- [x] prevent the user from starting bamm refinement without selected motifs.
- [x] download buttons for motifs on peng result page need fixing
- [x] BaMM workflow: ValueError: invalid literal for int() with base 10: 'motif' - because of newly introduced header?
- [x] replace former de-novo workflow on the main page with the new seed-refine workflow
- [x] rework populate_example.py for new database design
- [x] PengToBamm page should not show PEnG job_id as job_name - user should be able to choose the job_name
- [x] fix plots in BaMM result page
- [x] peng module problem: temp_dir seems to be unset - therefore several instances write on the same directory named `temp` simultaneously
- [x] Peng->Bamm job shows wrong job id in "Submission Successful" view  

## Important Changes
- [x] add multiple database support - (Christian)
- [x] add logging configuration - (Christian)
- [x] add header to peng result page: “Select Motifs for Higher Order Refinement”
- [x] replace MEME plotter with our own plotting script for consistency on peng result page
- [x] PEnG settings is blank on peng result page
- [x] peng results table should not have # sites (not very accurate), but instead AUSFC scoring
- [x] BaMM results coming from PEnG should have a link to the peng results
- [x] PengToBamm submit: do not give the user the option to choose reverse complement, hide the sequence set; use a foreign key to PEnGJoB in BaMMJob table to access this data from the PEnG job
- [ ] validate input of MMCompare
- [x] Overwork design of html pages, specifically results pages.
- [ ] Add error handling in case the task chain fails.

## Nice-to-have Changes
- [x] put original meme file in “Download all” archive on peng result page
- [ ] add MMcompare annotation to peng result page
- [ ] cleaning up unused classes/functions
- [ ] put job directories in `${MEDIA_DB}/jobs/uuid`, move log file into this directory.
- [ ] implement bamm tools as command line modules
## Changes after everything else is done
- [ ] cleanup directory and file name mess 

# How to work with the existing code
To avoid more nasty problems with the git history, here are the guidelines how to contribute code:
- fork the webserver repository, do not work directly on the soedinglab repository
- derive a feature branch from the `new_peng_workflow` remote branch in your fork
- commit your changes to your derived branch
- when you are done, open a pull request of your feature branch against `new_peng_workflow` in the soedinglab repository
